packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx

```diff
@@ -262,6 +266,17 @@ export const MetricPeekModal: FC<Props> = ({ opened, onClose }) => {
         >
             <Modal.Overlay />
             <Modal.Content sx={{ overflow: 'hidden' }} radius={12} w="100%">
+                <LoadingOverlay
+                    visible={
+                        metricQuery.isLoading || metricResultsQuery.isLoading
+                    }
+                    overlayBlur={2}
+                    loaderProps={{
+                        size: 'md',
+                        color: 'dark',
+                        variant: 'dots',
+                    }}
+                />
```

Comment:
this loading overlay only shows when the user opens the modal and its loading the metric info and the query for the first time. 

Then, on metric visualization ,it will show the loading spinner when fetching.